posts/2024-07-10-rln-v2-stateless-light-clients.mdx

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+name: 'Improved rate limiting and stateless light clients in The Waku Network'
+title: 'Improved rate limiting and stateless light clients in The Waku Network'
+date: 2024-07-10 00:00:00
+authors: []
+published: true
+slug: rln-v2-stateless-light-clients
+categories: waku, progress
+image: /img/black-waku-logo-with-name.png
+---
+
+> **TLDR**: We have just shipped two important features to The Waku Network that offer more granularity for RLN rate-limiting and add support for stateless light clients.
+
+# Introduction to The Waku Network
+
+Waku launched The Waku Network last [December 2023](https://thedefiant.io/news/press-releases/waku-launches-first-decentralised-privacy-preserving-dos-protections-for-peer-to-peer-messaging). A first-of-its-kind decentralized network focused on privacy and censorship resistance, running an opinionated instance of the Waku protocol with the following features:
+
+- Generalized messaging network, that allows point-to-point or broadcast communications, preserving the identity of the sender, not linking its messages to any identity or IP address.
+- A decentralized and permissionless network acting as a common infrastructure, not controlled by a single actor and open to anyone, both users and operators.
+- Rate-limited to ensure a fair usage of resources. It uses RLN (Rate Limiting Nullifiers) that leverage zero-knowledge cryptography to guarantee users adhere to a rate limit without compromising their privacy. Anyone can register in the smart contract, currently deployed in Sepolia testnet, by paying a small fee.
+- Support for light clients, defined as resource-restricted devices that interact with the protocol without participating in message routing.
+- Incentivized, both non-economic (reputation mechanism already implemented) and economic (crypto incentives currently under research).
+
+Over the last months, we detected a few limitations introduced by the usage of RLN for rate limiting, and we have been working hard to address them. As a result, we are proud to announce that the following new features are live in The Waku Network:
+
+> We are excited to announce that last 3rd of July 2024, we introduced the following features, part of release [v0.30.1](https://github.com/waku-org/nwaku/releases/tag/v0.30.1).
+> 
+- **Better rate-limiting:** The previous rate-limiting mechanism (RLN-V1) only allowed to rate limit `1` message every `y` seconds. This wasn't practical and it's why we have just shipped RLN-V2, which allows nodes to rate limit `x` messages every `y` seconds.
```

Comment:
Should we perhaps mention that `y` is fixed for a given SC deployment? In other words it's not quite a membership variable like `x` (just in case we need to contrast with RLN-V3 in future).